Src/AutoMoq/MockType.cs

```diff
@@ -68,11 +68,14 @@ internal static Type GetMockedType(this Type type)
             where TMock : class
         {
             return setup.Returns(() =>
-                {
-                    var result = (TResult) context.Resolve(typeof (TResult));
-                    setup.Returns(result);
-                    return result;
-                });
+            {
+                var specimen = context.Resolve(typeof (TResult));
+                TResult result = specimen is TResult ? (TResult) specimen : default(TResult);

```

Comment:
Well, this was a rather contrived example, I don't think I would ever want to do that... 

In fact, I've never had the need to use `OmitSpecimen` directly - I've only used it through DSL customizations (`Customize<Test>(c => c.Without(...))`) which only affect concrete classes, not interfaces..

It makes sense to want to omit a property of a concrete class, because the setter may have unwanted side-effects, but I can't think of a scenario where you'd want to omit the setup of a method. I don't think this scenario will come up often, if at all.